r/R/compute.R

```diff
@@ -307,3 +307,157 @@ cast_options <- function(safe = TRUE, ...) {
   )
   modifyList(opts, list(...))
 }
+
+#' Register user-defined functions
+#'
+#' These functions support calling R code from query engine execution
+#' (i.e., a [dplyr::mutate()] or [dplyr::filter()] on a [Table] or [Dataset]).
+#' Use [arrow_scalar_function()] to define an R function that accepts and
+#' returns R objects; use [arrow_advanced_scalar_function()] to define a
+#' lower-level function that operates directly on Arrow objects.
```

Comment:
Do you want to document here or somewhere else the restrictions on scalar functions?  In particular:

 * They should be stateless
 * The size of the output array must match the size of the input array (one output per row)